## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 493943`
Commit: `5a9254f81ab943ec32291d4dd19e793d94650587`

---
### `aarch64-darwin`
<details>
  <summary>:x: 13 packages failed to build:</summary>
  <ul>
    <li>gpodder</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>mkchromecast</li>
    <li>mkchromecast.dist</li>
    <li>pinchflat</li>
    <li>python313Packages.mpv</li>
    <li>python313Packages.mpv.dist</li>
    <li>python314Packages.mpv</li>
    <li>python314Packages.mpv.dist</li>
    <li>switchfin</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 82 packages built:</summary>
  <ul>
    <li>ani-cli</li>
    <li>auto-editor</li>
    <li>bilibili-tui</li>
    <li>botamusique</li>
    <li>catt</li>
    <li>catt.dist</li>
    <li>cplay-ng</li>
    <li>cplay-ng.dist</li>
    <li>curseradio</li>
    <li>curseradio.dist</li>
    <li>dmlive</li>
    <li>dra-cla</li>
    <li>feishin</li>
    <li>ff2mpv</li>
    <li>ff2mpv-go</li>
    <li>gallery-dl</li>
    <li>gallery-dl.dist</li>
    <li>gonic</li>
    <li>gophertube</li>
    <li>gtk-pipe-viewer</li>
    <li>gtk-pipe-viewer.devdoc</li>
    <li>invidtui</li>
    <li>jellyfin-tui</li>
    <li>jellytui</li>
    <li>kodiPackages.sendtokodi</li>
    <li>moodle-dl</li>
    <li>moodle-dl.dist</li>
    <li>mopidy-youtube (mopidyPackages.mopidy-youtube)</li>
    <li>mopidy-youtube.dist (mopidyPackages.mopidy-youtube.dist)</li>
    <li>mov-cli</li>
    <li>mov-cli.dist</li>
    <li>mpv</li>
    <li>mpvScripts.mpv-playlistmanager</li>
    <li>parrot</li>
    <li>persepolis</li>
    <li>pipe-viewer</li>
    <li>pipe-viewer.devdoc</li>
    <li>previewqt</li>
    <li>python313Packages.bgutil-ytdlp-pot-provider</li>
    <li>python313Packages.bgutil-ytdlp-pot-provider.dist</li>
    <li>python313Packages.tubeup</li>
    <li>python313Packages.tubeup.dist</li>
    <li>python313Packages.yark</li>
    <li>python313Packages.yark.dist</li>
    <li>yt-dlp (python313Packages.yt-dlp)</li>
    <li>yt-dlp-light (python313Packages.yt-dlp-light)</li>
    <li>yt-dlp-light.dist (python313Packages.yt-dlp-light.dist)</li>
    <li>yt-dlp.dist (python313Packages.yt-dlp.dist)</li>
    <li>python314Packages.bgutil-ytdlp-pot-provider</li>
    <li>python314Packages.bgutil-ytdlp-pot-provider.dist</li>
    <li>python314Packages.tubeup</li>
    <li>python314Packages.tubeup.dist</li>
    <li>python314Packages.yark</li>
    <li>python314Packages.yark.dist</li>
    <li>python314Packages.yt-dlp</li>
    <li>python314Packages.yt-dlp-light</li>
    <li>python314Packages.yt-dlp-light.dist</li>
    <li>python314Packages.yt-dlp.dist</li>
    <li>radio-cli</li>
    <li>shira</li>
    <li>shira.dist</li>
    <li>somafm-cli</li>
    <li>spotdl</li>
    <li>spotdl.dist</li>
    <li>subtitleedit</li>
    <li>szurubooru.server</li>
    <li>szurubooru.server.dist</li>
    <li>tomato-c</li>
    <li>video-downloader</li>
    <li>wtwitch</li>
    <li>yewtube</li>
    <li>yewtube.dist</li>
    <li>youtube-tui</li>
    <li>ytcc</li>
    <li>ytcc.dist</li>
    <li>ytdl-sub</li>
    <li>ytdl-sub.dist</li>
    <li>ytdownloader</li>
    <li>ytfzf</li>
    <li>ytmdl</li>
    <li>ytmdl.dist</li>
    <li>ytui-music</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>gpodder</summary>
<pre>    
                    if sys.platform == &quot;darwin&quot; and port == 5000:
                        print(
                            &quot;On macOS, try searching for and disabling&quot;
                            &quot; &#x27;AirPlay Receiver&#x27; in System Settings.&quot;,
                            file=sys.stderr,
                        )
    
&gt;               sys.exit(1)
E               SystemExit: 1

/nix/store/2kkyapgv4v9h0zlx4i94fsl81xp6681s-python3.13-werkzeug-3.1.5/lib/python3.13/site-packages/werkzeug/serving.py:782: SystemExit
---------------------------- Captured stderr setup -----------------------------
Operation not permitted
=========================== short test summary info ============================
ERROR tests/test_feedcore.py::test_easy - SystemExit: 1
ERROR tests/test_feedcore.py::test_redirect - SystemExit: 1
ERROR tests/test_feedcore.py::test_redirect_loop - SystemExit: 1
ERROR tests/test_feedcore.py::test_temporary_error_retry - SystemExit: 1
============================== 4 errors in 0.84s ===============================</pre>
</details>
<details>
<summary>pinchflat</summary>
<pre>    (public_key 1.20.1) pubkey_os_cacerts.erl:301: :pubkey_os_cacerts.conv_error_reason(:no_cacerts_found)
    (public_key 1.20.1) pubkey_os_cacerts.erl:56: :pubkey_os_cacerts.get/0
    (esbuild 0.10.0) lib/esbuild/npm_registry.ex:101: Esbuild.NpmRegistry.do_fetch/1
    (esbuild 0.10.0) lib/esbuild/npm_registry.ex:67: Esbuild.NpmRegistry.fetch_file!/2
    (esbuild 0.10.0) lib/esbuild/npm_registry.ex:46: Esbuild.NpmRegistry.fetch_package!/2
    (esbuild 0.10.0) lib/esbuild.ex:241: Esbuild.install/0
    (elixir 1.18.4) lib/task/supervised.ex:101: Task.Supervised.invoke_mfa/2
Function: #Function&lt;1.110144940/0 in Esbuild.install&gt;
    Args: []

10:44:54.623 [debug] Downloading esbuild from https://registry.npmjs.org/@esbuild/darwin-arm64/0.17.11
** (ErlangError) Erlang error: :enoent
    :erlang.open_port({:spawn_executable, ~c&quot;/nix/var/nix/b/3pkc1hds6higdh6wlcd9whg505/source/_build/esbuild-darwin-arm64&quot;}, [:use_stdio, :stderr_to_stdout, {:env, [{~c&quot;NODE_PATH&quot;, ~c&quot;/nix/var/nix/b/3pkc1hds6higdh6wlcd9whg505/source/deps&quot;}]}, {:cd, &quot;/nix/var/nix/b/3pkc1hds6higdh6wlcd9whg505/source/assets&quot;}, :exit_status, :binary, :hide, {:args, [&quot;js/app.js&quot;, &quot;--bundle&quot;, &quot;--target=es2017&quot;, &quot;--outdir=../priv/static/assets&quot;, &quot;--external:/fonts/*&quot;, &quot;--external:/images/*&quot;, &quot;--minify&quot;]}])
    (elixir 1.18.4) lib/system.ex:1131: System.do_cmd/3
    (esbuild 0.10.0) lib/esbuild.ex:184: Esbuild.run/2
    (esbuild 0.10.0) lib/mix/tasks/esbuild.ex:56: Mix.Tasks.Esbuild.install_and_run/1
    (mix 1.18.4) lib/mix/task.ex:495: anonymous fn/3 in Mix.Task.run_task/5
    (elixir 1.18.4) lib/enum.ex:987: Enum.&quot;-each/2-lists^foreach/1-0-&quot;/2
    (mix 1.18.4) lib/mix/task.ex:495: anonymous fn/3 in Mix.Task.run_task/5
    (mix 1.18.4) lib/mix/cli.ex:107: Mix.CLI.run_task/2</pre>
</details>
<details>
<summary>python313Packages.mpv</summary>
<pre>FAILED tests/test_mpv.py::TestStreams::test_play_context - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_python_stream - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_python_stream_exception - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_stream_open_exception - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_stream_open_forward - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_log_handler - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_event - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_event_shutdown - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_event_overflow - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_negative - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_positive - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_shutdown - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_shutdown - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::CommandTests::test_async_command - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::CommandTests::test_loadfile_with_subtitles - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::CommandTests::test_sub_add - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::RegressionTests::test_instance_method_property_observer - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::RegressionTests::test_unobserve_property_runtime_error - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::RegressionTests::test_wait_for_property_concurrency - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
========================= 43 failed, 4 passed in 8.83s =========================</pre>
</details>
<details>
<summary>python314Packages.mpv</summary>
<pre>FAILED tests/test_mpv.py::TestStreams::test_play_context - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_python_stream - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_python_stream_exception - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_stream_open_exception - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestStreams::test_stream_open_forward - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_log_handler - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_event - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_event_shutdown - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_event_overflow - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_negative - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_positive - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_property_shutdown - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::TestLifecycle::test_wait_for_shutdown - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::CommandTests::test_async_command - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::CommandTests::test_loadfile_with_subtitles - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::CommandTests::test_sub_add - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::RegressionTests::test_instance_method_property_observer - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::RegressionTests::test_unobserve_property_runtime_error - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
FAILED tests/test_mpv.py::RegressionTests::test_wait_for_property_concurrency - FileNotFoundError: [Errno 2] No such file or directory: &#x27;Xvfb&#x27;
========================= 43 failed, 4 passed in 9.24s =========================</pre>
</details>
<details>
<summary>switchfin</summary>
<pre>[ 93%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/presenter.cpp.o
[ 94%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/recycling_grid.cpp.o
[ 94%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/recyling_video.cpp.o
[ 95%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/search_list.cpp.o
[ 96%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/selector_cell.cpp.o
[ 96%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/svg_image.cpp.o
[ 97%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/text_box.cpp.o
[ 97%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/video_card.cpp.o
[ 98%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/video_profile.cpp.o
[ 98%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/video_progress_slider.cpp.o
[ 99%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/video_source.cpp.o
1 warning generated.
[ 99%] Building CXX object CMakeFiles/Switchfin.dir/app/src/view/video_view.cpp.o
Copying OS X content Switchfin.app/Contents/Resources/app.icns
[100%] Linking CXX executable Switchfin.app/Contents/MacOS/Switchfin
ld: library not found for -lstdc++fs
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [CMakeFiles/Switchfin.dir/build.make:1187: Switchfin.app/Contents/MacOS/Switchfin] Fehler 1
make[1]: *** [CMakeFiles/Makefile2:238: CMakeFiles/Switchfin.dir/all] Fehler 2
make: *** [Makefile:136: all] Fehler 2</pre>
</details>
</details>

--- a/pkgs/by-name/yt/yt-dlp/package.nix
+++ b/pkgs/by-name/yt/yt-dlp/package.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenvNoCC,
   python3Packages,
   atomicparsley,
   deno,
@@ -13,6 +14,7 @@
   javascriptSupport ? true,
   rtmpSupport ? true,
   withAlias ? false, # Provides bin/youtube-dl for backcompat
+  withSecretStorage ? !stdenvNoCC.hostPlatform.isDarwin,
   nix-update-script,
 }:
 
@@ -51,7 +53,10 @@ python3Packages.buildPythonApplication rec {
   ];
 
   # expose optional-dependencies, but provide all features
-  dependencies = lib.concatAttrValues optional-dependencies;
+  dependencies =
+    optional-dependencies.default
+    ++ optional-dependencies.curl-cffi
+    ++ lib.optionals withSecretStorage optional-dependencies.secretstorage;
 
   optional-dependencies = {
     default = with python3Packages; [

